idx/memory/memory.go

```diff
@@ -883,20 +883,27 @@ func (m *MemoryIdx) find(orgId int, pattern string) ([]*Node, error) {
 		}
 	}
 	var startNode *Node
+	var branch string
 	if pos == 0 {
 		//we need to start at the root.
 		log.Debug("memory-idx: starting search at the root node")
 		startNode = tree.Items[""]
 	} else {
-		branch := strings.Join(nodes[0:pos], ".")
+		branch = strings.Join(nodes[0:pos], ".")
 		log.Debug("memory-idx: starting search at branch %s", branch)
 		startNode, ok = tree.Items[branch]
 		if !ok {
 			log.Debug("memory-idx: branch %s does not exist in the index for orgId %d", branch, orgId)
-			return results, nil
+			return nil, nil
 		}
 	}
 
+	if startNode == nil {
```

Comment:
when the assignment of `startNode` on line `:894` fails because the searched `branch` is not present, then `ok` should be `false` and the error handler on `:895` would return. so the only case how this `startNode` could be `nil` is if the assignment in `:890` fails or not? so why not just add an `if !ok` check after line `:890` instead

Reply:
or is the goal to check if a value of `nil` was actually stored in `tree.Items`? then that makes sense, although i think it should be treated as a separate error case from the case where `""` does not exist in `tree.Items`

Reply:
`!ok` means an entry wasn't found in the map (and the returned value is then nil in that case).
however it is also possible there was an entry in the map that is nil (e.g. `ok` is true, returned value is a nil pointer).
i want to protect against all scenarios. (in fact i have a suspicion there might be a nilpointer in the map)

Reply:
I agree with @replay.  An index that is empty is not corrupt.  We should just return `nil, nil` after line `:890`

Reply:
i just changed the code to allow for the root node to not exist. and still treat found-but-nil nodes as corruption case. 3f78925065fe210cc85611b78983be962d55d0bb